lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@zeroedin-bill I wouldn't imagine most would like the idea of requiring double-connections.

What PHP version was `PGSQL_LIBPQ_VERSION ` introduced? That's exactly what I was asking for earlier (to use client_encoding for PG >= 9.1 or fallback to SET NAMES). BUT, only if that constant is reliably available.

I'll look into it... that would be an easy addition to my current state.

Originally I had asked if there was a way to do a version compare -- and only use this method for 9.1+. I wasn't aware of the PGSQL_LIBPQ_VERSION constant. I like this.